### ISSUE TITLE
All output files should use the name of the cproject.yml file as base name.

### ISSUE THREAD

**Issue by ReinhardKeil** [Open-CMSIS-Pack/devtools#831]
This is the expected output for a Hello project

```
Hello.axf
Hello.axf.map
Hello.bin
Hello.hex
Hello.Debug+AVH.clog    // log file with current build/target-type
```

When a build process starts, the build tool should check if there is a log file with the current build/target-type.   If this does not exist, the build output artifacts cannot be reused and to ensure a proper rebuild, the build tool could simply delete all files with the name `<project-name>.*`, in the example above `Hello.*`.  This would ensure that a change of build/target-type in a **build context manager** would remove inconsistent artifacts.

**Comment by brondani** [comment]
The `project name` as default output name has been implemented in the PR https://github.com/Open-CMSIS-Pack/devtools/pull/836.
The handling of rebuilding was not implemented. Conceptually this should be a CMake task, I would recommend to avoid overwriting build artifacts.

**Comment by ReinhardKeil** [comment]
Great this is all we need for now.  Once the tool is available, I suggest to try this:
https://github.com/Open-CMSIS-Pack/devtools/issues/826#issuecomment-1508586239

**Comment by jkrech** [comment]
see: CMSIS-Toolbox 1.6.0

**Comment by ReinhardKeil** [comment]
Fundamental problem is fixed in 1.6.0, I still suggest to implement #855
